### PR TITLE
Bioenergy

### PR DESCRIPTION
**Pull request by kvcalvin**
This adds the ability to include bioenergy in future scenarios, following the approach in GCAM4. To do this, I had to make share weights a period vector. Note that there aren't any tests for bioenergy yet, because I'm not sure what/how to test. If you have suggestions, let me know.

### CODE REVIEW

**Comment by rplzzz** [comment]
It looks like three of the existing tests failed:
```
  ══ testthat results  ═══════════════════════════════════════════════════════════
  OK: 102 SKIPPED: 0 FAILED: 3
  1. Error: yield expectation calculation works (@test_expectations.R#10)
  2. Failure: land cover matches reference values (@test_model_output.R#235)
  3. Failure: land cover matches reference values (@test_reference.R#49)
```
Should these changes have changed those results?

File: R/ag_production_technology.R [hunk 1]
```diff
@@ -57,7 +57,10 @@ AgProductionTechnology_initCalc <- function(aLandLeaf, aPeriod, aScenarioInfo) {
       # Adjust last period's yield by tech change
       aLandLeaf$mYield[aPeriod] <- preYield * ((1 + aLandLeaf$mAgProdChange[[aPeriod]]) ^ timestep)
     }
-  } else {
+  } else if (length(aLandLeaf$mYield) > 0) {
```
Comment (rplzzz) [inline R/ag_production_technology.R]: FWIW, the combination of convoluted control paths and dense formatting makes this entire function very hard to understand.  Generally speaking, when you find yourself writing something like this, you should be asking if there is some kind of refactoring you could do.

File: tests/testthat/test_model_output.R [hunk 2]
```diff
@@ -189,56 +189,6 @@ test_that("land area doesn't change over time", {
 
 })
 
-
-test_that("land cover matches reference values", {
```
Comment (rplzzz) [inline tests/testthat/test_model_output.R]: Wasn't this testing that when we ran a configuration where we knew the answer, we actually got the answer we were expecting?  If so, is there an equivalent test we can do to verify correctness?

Comment (kvcalvin) [reply tests/testthat/test_model_output.R]: Yes, but that test appeared in two different test files. I removed this one and left it in the other.

**Comment by rplzzz** [comment]
Looks like this one is set to go, so I'm going to go ahead with the merge.

**Pull request merged**